windows/src/engine/keyman32/keyboardoptions.cpp

```diff
@@ -230,63 +230,57 @@ void LoadKeyboardOptionsREGCore(LPINTKEYBOARDINFO kp, km_kbp_state* const state)
   IntLoadKeyboardOptionsCore(REGSZ_KeyboardOptions, kp, state);
 }
 
-BOOL IntLoadKeyboardOptionsCore(LPCSTR key, LPINTKEYBOARDINFO kp, km_kbp_state* state)
+BOOL IntLoadKeyboardOptionsCore(LPCSTR key, LPINTKEYBOARDINFO kp, km_kbp_state* const state)
 {
   assert(key != NULL);
   assert(kp != NULL);
 
   RegistryReadOnly r(HKEY_CURRENT_USER);
-  if (r.OpenKeyReadOnly(REGSZ_KeymanActiveKeyboards) && r.OpenKeyReadOnly(kp->Name) && r.OpenKeyReadOnly(key))
-  {
-    WCHAR buf[256];
-    int n = 0;
-    // Get the list of default options to determine size of list
-    const km_kbp_keyboard_attrs* keyboardAttrs;
-    km_kbp_status err_status = km_kbp_keyboard_get_attrs(kp->lpCoreKeyboard, &keyboardAttrs);
-    if (err_status != KM_KBP_STATUS_OK) {
-      SendDebugMessageFormat(
-          0, sdmKeyboard, 0, "LoadKeyboardOptionsREGCore: km_kbp_keyboard_get_attrs failed with error status [%d]", err_status);
-      return FALSE;
-    }
-    size_t listSize = km_kbp_options_list_size(keyboardAttrs->default_options);
-    km_kbp_option_item* keyboardOpts = new  km_kbp_option_item[listSize+1];
-
-    while ((r.GetValueNames(buf, sizeof(buf) / sizeof(buf[0]), n)) && (n < (int)listSize))
-    {
-      buf[255] = 0;
-      WCHAR val[256];
 
-      if (r.ReadString(buf, val, sizeof(val) / sizeof(val[0])) && val[0])
-      {
-        val[255] = 0;
-        keyboardOpts[n].scope = KM_KBP_OPT_KEYBOARD;
-
-        km_kbp_cp* cp = new km_kbp_cp[wcslen(buf) + 1];
-        wcscpy_s(reinterpret_cast<LPWSTR>(cp), wcslen(buf) + 1, buf);
-        keyboardOpts[n].key = cp;
+  if (!(r.OpenKeyReadOnly(REGSZ_KeymanActiveKeyboards) && r.OpenKeyReadOnly(kp->Name) && r.OpenKeyReadOnly(key))) {
```

Comment:
```suggestion
  BOOL hasData = r.OpenKeyReadOnly(REGSZ_KeymanActiveKeyboards) && r.OpenKeyReadOnly(kp->Name) && r.OpenKeyReadOnly(key);
```

We need to be setting to default if no data available, for consistency, right?